yolov3_deepsort.py

```diff
@@ -28,82 +35,126 @@ def __init__(self, cfg, args):
         self.deepsort = build_tracker(cfg, use_cuda=use_cuda)
         self.class_names = self.detector.class_names
 
-
     def __enter__(self):
-        assert os.path.isfile(self.args.VIDEO_PATH), "Error: path error"
         self.vdo.open(self.args.VIDEO_PATH)
         self.im_width = int(self.vdo.get(cv2.CAP_PROP_FRAME_WIDTH))
         self.im_height = int(self.vdo.get(cv2.CAP_PROP_FRAME_HEIGHT))
+        self.total_frames = int(cv2.VideoCapture.get(self.vdo, cv2.CAP_PROP_FRAME_COUNT))
+        video_details = {'frame_width': self.im_width,
+                         'frame_height': self.im_height,
+                         'frame_rate': self.args.write_fps,
+                         'video_name': self.args.input}
+        self.json_logger.add_video_details(**video_details)
 
         if self.args.save_path:
-            fourcc =  cv2.VideoWriter_fourcc(*'MJPG')
-            self.writer = cv2.VideoWriter(self.args.save_path, fourcc, 20, (self.im_width,self.im_height))
+            fourcc = cv2.VideoWriter_fourcc(*'MJPG')
+            filename, extension = splitext(basename(self.args.VIDEO_PATH))
+            self.output_file = join(self.args.save_path, f'{filename}.avi')
+            self.json_output = join(self.args.save_path, f'{filename}.json')
+            if not isdir(dirname(self.json_output)):
+                os.makedirs(dirname(self.json_output))
+            self.writer = cv2.VideoWriter(self.args.output_file, fourcc, 20, (self.im_width, self.im_height))
 
         assert self.vdo.isOpened()
         return self
 
-    
     def __exit__(self, exc_type, exc_value, exc_traceback):
         if exc_type:
             print(exc_type, exc_value, exc_traceback)
-        
 
     def run(self):
         idx_frame = 0
-        while self.vdo.grab(): 
-            idx_frame += 1
-            if idx_frame % self.args.frame_interval:
-                continue
-
-            start = time.time()
-            _, ori_im = self.vdo.retrieve()
-            im = cv2.cvtColor(ori_im, cv2.COLOR_BGR2RGB)
-
-            # do detection
-            bbox_xywh, cls_conf, cls_ids = self.detector(im)
-            if bbox_xywh is not None:
-                # select person class
-                mask = cls_ids==0
-
-                bbox_xywh = bbox_xywh[mask]
-                bbox_xywh[:,3:] *= 1.2 # bbox dilation just in case bbox too small
-                cls_conf = cls_conf[mask]
-
-                # do tracking
-                outputs = self.deepsort.update(bbox_xywh, cls_conf, im)
+        pbar = tqdm(total=self.total_frames + 1)
+        while self.vdo.grab():
+            if idx_frame % args.frame_interval == 0:

```

Comment:
I used frame_intervals as a way to skip the process for faster results.
if we use frame_intervals more than 7, the tracker performance decreases.